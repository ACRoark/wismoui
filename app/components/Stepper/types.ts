import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
