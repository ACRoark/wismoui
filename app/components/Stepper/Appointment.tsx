import React from 'react';
import { FormattedMessage } from 'react-intl';

import { SlotType } from 'types';

import getAppointmentType from './getAppointmentType';
import getTime from './getTime';
import messages from './messages';

interface IAppointmentProps {
  product: string;
  slotType: SlotType;
}

const Appointment: React.FC<IAppointmentProps> = (props: IAppointmentProps): React.ReactElement => {
  const { product, slotType } = props;

  const time = getTime(slotType);
  const serviceType = getAppointmentType(product);

  return (
    <>
      {product && slotType && slotType !== 'AN' && (
        <div>
          <FormattedMessage {...messages[serviceType]} values={{ window: time }} />
        </div>
      )}
    </>
  );
};

export default Appointment;
