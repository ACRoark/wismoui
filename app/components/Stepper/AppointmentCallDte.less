@import (reference) '../../styles/themes/index.less';

@appointment-call-dte-cls-prefix: ~'@{dte-prefix}-appointment-call-dte';

.@{appointment-call-dte-cls-prefix} {
  .message {
    color: @dte-red;
    font-weight: @dte-wismo-font-weight-semi-bold;
  }
}
