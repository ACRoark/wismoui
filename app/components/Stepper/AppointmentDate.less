@import (reference) '../../styles/themes/index.less';

@appointment-date-cls-prefix: ~'@{dte-prefix}-appointment-date';

.@{appointment-date-cls-prefix} {
  display: flex;
  flex-direction: row;
}
