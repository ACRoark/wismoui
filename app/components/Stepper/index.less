@import (reference) '../../styles/themes/index.less';

@step-prefix-cls: ~'@{dte-prefix}-stepper';

.@{step-prefix-cls} {
  *::before,
  *::after {
    box-sizing: unset !important;
  }

  .@{dte-prefix}-step {
    box-sizing: unset !important;
    display: flex;
    position: relative;
    &:last-child {
      .content {
        min-height: unset;
      }
      .tail {
        height: 0;
      }
    }

    a {
      padding-left: 10px;
    }

    .container {
      box-sizing: unset !important;
      display: flex;
    }

    .content {
      min-height: 100px;
    }

    .description {
      font-size: 1rem;
      line-height: 1.424rem;
      margin-bottom: 25px;
      padding: 10px 0;
    }

    .half-1 {
      height: 30%;
    }

    .half-2 {
      height: 70%;
    }

    .icon {
      box-sizing: unset !important;
      padding: 2px;
      display: flex;
      justify-content: center;
      align-items: center;
      border-radius: 50%;
      height: 24px;
      width: 24px;
      transition: background-color 0.3s, border-color 0.3s;
    }

    .icon-wrapper {
      padding: 0 15px 10px 10px;

      .active {
        background: @dte-white;
        color: @dte-white;
        border: 2px solid @dte-blue;
      }
      .completed {
        background: @dte-blue;
        color: @dte-white;
        border: 2px solid @dte-blue;
      }
      .error {
        background: @dte-white;
        color: @dte-red;
        border: 2px solid @dte-red;
      }
      .pending {
        background: @dte-white;
        color: @dte-white;
        border: 2px solid @dte-gray-300;
      }
    }

    .tail {
      box-sizing: unset !important;
      display: block;
      position: absolute;
      top: 0;
      left: 25px;
      padding: 32px 0 6px;
      width: 2px;
      height: 100%;
    }

    .title {
      font-size: @dte-wismo-font-size-3;
      line-height: 2.027rem;
    }

    //Status styling

    .active {
      .half-1 {
        background: @dte-blue;
      }
      .half-2 {
        background: @dte-gray-300;
      }
      .content {
        color: @dte-gray-600;
      }
    }

    .completed {
      .half-1,
      .half-2 {
        background: @dte-blue;
      }
      .content {
        color: @dte-gray-700;
      }
    }

    .error {
      .half-1 {
        background: @dte-red;
      }
      .half-2 {
        background: @dte-gray-300;
      }

      .title {
        color: @dte-red;
      }

      .verification-content-error {
        color: @dte-red;
        margin-bottom: 10px;
      }

      .verification-content-error:last-child {
        margin-bottom: 0;
      }

      .view-details-link {
        font-size: @dte-wismo-font-size-0;
        padding: unset;
        padding-left: 13px;
      }

      @media @mobile {
        .view-details-link {
          display: block;
          padding: unset;
          padding-top: 10px;
        }
      }
    }

    .pending {
      .half-1,
      .half-2 {
        background: @dte-gray-300;
      }
      .content {
        color: @dte-gray-600;
      }
    }

    .single-product {
      display: flex;
    }
  }
}
