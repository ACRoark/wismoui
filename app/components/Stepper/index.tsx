import React, { FC, ReactElement } from 'react';

import useFlags from 'hooks/useFlags';
import IOrderRequest from 'types/IOrderRequest';
import getSteps from './getSteps';
import Step from './Step';
import { IStepDescription } from './types';

import './index.less';

interface IStepperProps {
  orderCreatedDate: string;
  serviceRequest: IOrderRequest;
}

const Stepper: FC<IStepperProps> = (props: IStepperProps): ReactElement => {
  const { bug3665, bug3883, us1599, us1946, us3273 } = useFlags();
  const { orderCreatedDate, serviceRequest } = props;

  const options = {
    bug3665,
    bug3883,
    us1599,
    us1946,
    us3273,
  };

  return (
    <div className="dte-wismo-stepper">
      {getSteps(orderCreatedDate, serviceRequest, options).map(
        (step: IStepDescription): ReactElement => (
          <Step key={step.key} {...step} />
        ),
      )}
    </div>
  );
};

export default Stepper;
