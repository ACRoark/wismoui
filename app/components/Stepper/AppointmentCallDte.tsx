import React from 'react';
import { FormattedMessage } from 'react-intl';

import DteContactNumber from 'components/DteContactNumber';
import { ProductType } from 'types';

import getAppointmentType from './getAppointmentType';
import messages from './messages';

import './AppointmentCallDte.less';

interface IAppointmentCallDteProps {
  product: ProductType;
}

const AppointmentCallDte: React.FC<IAppointmentCallDteProps> = (
  props: IAppointmentCallDteProps,
): React.ReactElement => {
  const { product } = props;

  const serviceType = getAppointmentType(product);

  return (
    <div className="dte-wismo-appointment-call-dte">
      <div>
        <FormattedMessage {...messages[serviceType]} values={{ window: null }} />
      </div>
      <span className="message">
        <FormattedMessage
          {...messages.CALL_DTE}
          values={{
            phoneNumber: <DteContactNumber />,
          }}
        />
      </span>
    </div>
  );
};

export default AppointmentCallDte;
