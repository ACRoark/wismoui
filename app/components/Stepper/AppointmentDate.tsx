import React from 'react';

import DateStamp from 'components/DateStamp';
import EditServiceDateLink from 'components/EditServiceDateLink';

import './AppointmentDate.less';

interface IAppointmentDateProps {
  date: string;
}

const AppointmentDate: React.FC<IAppointmentDateProps> = (props: IAppointmentDateProps): React.ReactElement => {
  const { date } = props;

  return (
    <div className="dte-wismo-appointment-date">
      <DateStamp value={date} />
      <EditServiceDateLink wantDate={date} />
    </div>
  );
};

export default AppointmentDate;
