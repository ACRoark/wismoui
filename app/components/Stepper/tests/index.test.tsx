import React from 'react';
import { Provider } from 'react-redux';

import Stepper from 'components/Stepper';
import useFlags from 'hooks/useFlags';
import { ModalProvider } from 'providers/ModalProvider';
import { userIsAuthenticated } from 'security/utils';
import { httpClient } from 'services/httpClient';
import { createMockStore, createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { IOrderRequest, } from 'types';

jest.mock('hooks/useFlags');
jest.mock('security/utils');
jest.mock('services/httpClient');

const mockAuth = userIsAuthenticated as jest.MockedFunction<typeof userIsAuthenticated>;
const mockFlags = useFlags as jest.MockedFunction<typeof useFlags>;
const mockHttpClient = httpClient as jest.Mocked<typeof httpClient>;
const store = createMockStore();
const orderCreatedDate = '20200304';

const serviceRequest: IOrderRequest = {
  accountNumber: '001234567890',
  address: {
    city: 'Hobbiton',
    line1: '1 Bagshot Row',
    line2: null,
    state: 'TS',
    zip: '02890',
  },
  contactPhoneNumber: '7348675309',
  orderRequestType: 'MIMO_START',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
    {
      createdAtDate: '20200320',
      createdAtTime: '092215',
      orderRequestStatus: 'PROCESSED',
    },
    {
      createdAtDate: '20200321',
      createdAtTime: '223443',
      orderRequestStatus: 'SCHEDULED',
    },
  ],
  premiseId: 'premise123',
  products: [
    {
      productType: 'EFR_D1',
      serviceDate: '20200422',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '20200422',
        slotType: 'AM',
      },
      latestStatus: {
        createdAtDate: '20200321',
        createdAtTime: '223443',
        serviceOrderEventStatus: 'SCHEDULED_TODAY',
      },
      product: 'EFR_D1',
      serviceOrderId: 'abc123',
      serviceOrderCategory: null,
    },
    {
      appointment: {
        date: '20200422',
        slotType: 'PM',
      },
      latestStatus: {
        createdAtDate: '20200321',
        createdAtTime: '223443',
        serviceOrderEventStatus: 'SCHEDULED_TODAY',
      },
      product: 'GCC_GS_1H',
      serviceOrderCategory: null,
      serviceOrderId: 'abc123',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200320',
          createdAtTime: '092215',
        },
      ],
    },
  ],
  wantDate: '20200422',
};

const multipleBpemsServiceRequest: IOrderRequest = {
  accountNumber: '001234567890',
  address: {
    city: 'Hobbiton',
    line1: '1 Bagshot Row',
    line2: null,
    state: 'TS',
    zip: '02890',
  },
  contactPhoneNumber: '7348675309',
  orderRequestType: 'MIMO_START',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
    {
      createdAtDate: '20200320',
      createdAtTime: '092215',
      orderRequestStatus: 'NEEDS_VERIFICATION',
    },
  ],
  premiseId: 'premise123',
  products: [
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-22',
        slotType: 'AM',
      },
      latestStatus: {
        createdAtDate: '20200321',
        createdAtTime: '223443',
        serviceOrderEventStatus: 'SCHEDULED_TODAY',
      },
      product: 'EFR_D1',
      serviceOrderId: 'abc123',
      serviceOrderCategory: null,
    },
    {
      appointment: {
        date: '2020-04-22',
        slotType: 'PM',
      },
      latestStatus: {
        createdAtDate: '20200321',
        createdAtTime: '223443',
        serviceOrderEventStatus: 'SCHEDULED_TODAY',
      },
      product: 'GCC_GS_1H',
      serviceOrderCategory: null,
      serviceOrderId: 'abc123',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'ACTIVE',
          createdAtDate: '20200320',
          createdAtTime: '092215',
        },
      ],
    },
    {
      verificationCategory: 'R980',
      verificationEventId: '123124',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'PR',
          verificationEventStatus: 'ACTIVE',
          createdAtDate: '20200320',
          createdAtTime: '092215',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

const languages = ['en', 'es'];

describe('Stepper', (): void => {
  beforeEach((): void => {
    mockAuth.mockReturnValue(true);
    mockFlags.mockReturnValue({
      canChangeLanguage: true,
      developerMode: true,
    });
    mockHttpClient.post.mockReturnValue(new Promise((): string => ''));
  });

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<Provider store={store}><Stepper orderCreatedDate={orderCreatedDate} serviceRequest={serviceRequest} /></Provider>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should show correct steps and descriptions when language is ${language} and given a serivceRequest`, (): void => {
      const tree = createSnapshotWithIntl(
      <Provider store={store}>
        <Stepper orderCreatedDate={orderCreatedDate} serviceRequest={serviceRequest} />
      </Provider>, language
      );

      expect(tree).toMatchSnapshot();
    });

    it(`should show correct steps and description when language is ${language}, given a serviceRequest, and us1946 flag is true`, (): void => {
      mockAuth.mockReturnValue(true);
      mockFlags.mockReturnValue({
        bug3665: true,
        bug3883: true,
        canChangeLanguage: true,
        developerMode: true,
        us1946: true,
      });

      const tree = createSnapshotWithIntl(
        <Provider store={store}>
          <ModalProvider>
          <Stepper orderCreatedDate={orderCreatedDate} serviceRequest={multipleBpemsServiceRequest} />
          </ModalProvider>
        </Provider>, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
