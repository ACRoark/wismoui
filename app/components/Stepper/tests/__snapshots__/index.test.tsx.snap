// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should show correct steps and description when language is en, given a serviceRequest, and us1946 flag is true 1`] = `
<div
  className="dte-wismo-stepper"
>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Order Received
        </div>
        <div
          className="description"
        >
          03/04/20
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container error"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="error icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.002 11a1 1 0 112 0 1 1 0 01-2 0zM7.1 4.995a.905.905 0 111.8 0l-.35 3.507a.552.552 0 01-1.1 0L7.1 4.995z"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Verification
        </div>
        <div
          className="description"
        >
          <div
            className="verification-content-error"
          >
            Documentation Required
             - 
            Identity Verification
            <button
              className="view-details-link dte-wismo-button dte-wismo-button-link"
              data-track="mimo"
              data-track-action="click"
              data-track-detail="order tracker"
              onClick={[Function]}
            >
              View Details
            </button>
          </div>
          <div
            className="verification-content-error"
          >
            Pending
             - 
            Energy Resolution
            <button
              className="view-details-link dte-wismo-button dte-wismo-button-link"
              data-track="mimo"
              data-track-action="click"
              data-track-detail="order tracker"
              onClick={[Function]}
            >
              View Details
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Scheduled
        </div>
        <div
          className="description"
        >
          <div
            className="dte-wismo-appointment-date"
          >
            04/22/20
          </div>
          <div>
            Electric appointment 
            8 a.m. - 12 p.m.
          </div>
          <div>
            Natural Gas appointment 
            12 p.m. - 4:30 p.m.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container pending"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <i
          className="icon pending"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Completed 
        </div>
        <div
          className="description"
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper should show correct steps and description when language is es, given a serviceRequest, and us1946 flag is true 1`] = `
<div
  className="dte-wismo-stepper"
>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Orden recibida
        </div>
        <div
          className="description"
        >
          04/03/20
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container error"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="error icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.002 11a1 1 0 112 0 1 1 0 01-2 0zM7.1 4.995a.905.905 0 111.8 0l-.35 3.507a.552.552 0 01-1.1 0L7.1 4.995z"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Verificación
        </div>
        <div
          className="description"
        >
          <div
            className="verification-content-error"
          >
            Documentación Requerida
             - 
            Verificación de Identidad
            <button
              className="view-details-link dte-wismo-button dte-wismo-button-link"
              data-track="mimo"
              data-track-action="click"
              data-track-detail="order tracker"
              onClick={[Function]}
            >
              Ver Detalles
            </button>
          </div>
          <div
            className="verification-content-error"
          >
            Pendiente
             - 
            Resolución Energética
            <button
              className="view-details-link dte-wismo-button dte-wismo-button-link"
              data-track="mimo"
              data-track-action="click"
              data-track-detail="order tracker"
              onClick={[Function]}
            >
              Ver Detalles
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Programado
        </div>
        <div
          className="description"
        >
          <div
            className="dte-wismo-appointment-date"
          >
            22/04/20
          </div>
          <div>
            Cita electrica 
            8 a.m. - 12 p.m.
          </div>
          <div>
            Cita de gas natural 
            12 p.m. - 4:30 p.m.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container pending"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <i
          className="icon pending"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Terminado 
        </div>
        <div
          className="description"
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper should show correct steps and descriptions when language is en and given a serivceRequest 1`] = `
<div
  className="dte-wismo-stepper"
>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Order Received
        </div>
        <div
          className="description"
        >
          03/04/20
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Verification
        </div>
        <div
          className="description"
        >
          <span>
            Completed 
             
            03/20/20
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Scheduled
        </div>
        <div
          className="description"
        >
          <div
            className="dte-wismo-appointment-date"
          >
            04/22/20
          </div>
          <div>
            Electric appointment 
            8 a.m. - 12 p.m.
          </div>
          <div>
            Natural Gas appointment 
            12 p.m. - 4:30 p.m.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container pending"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <i
          className="icon pending"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Completed 
        </div>
        <div
          className="description"
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper should show correct steps and descriptions when language is es and given a serivceRequest 1`] = `
<div
  className="dte-wismo-stepper"
>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Orden recibida
        </div>
        <div
          className="description"
        >
          04/03/20
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Verificación
        </div>
        <div
          className="description"
        >
          <span>
            Terminado 
             
            20/03/20
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container completed"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <svg
          className="completed icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.854 3.646a.5.5 0 010 .708l-7 7a.5.5 0 01-.708 0l-3.5-3.5a.5.5 0 11.708-.708L6.5 10.293l6.646-6.647a.5.5 0 01.708 0z"
            fillRule="evenodd"
            stroke="currentColor"
            strokeWidth="1px"
          />
        </svg>
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Programado
        </div>
        <div
          className="description"
        >
          <div
            className="dte-wismo-appointment-date"
          >
            22/04/20
          </div>
          <div>
            Cita electrica 
            8 a.m. - 12 p.m.
          </div>
          <div>
            Cita de gas natural 
            12 p.m. - 4:30 p.m.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-step"
  >
    <div
      className="container pending"
    >
      <div
        className="tail"
      >
        <div
          className="half-1"
        />
        <div
          className="half-2"
        />
      </div>
      <div
        className="icon-wrapper"
      >
        <i
          className="icon pending"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="title"
        >
          Terminado 
        </div>
        <div
          className="description"
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;
